trunk/SUAVE/Methods/Aerodynamics/Common/Fidelity_Zero/Lift/generate_propeller_wake_distribution.py

```diff
@@ -0,0 +1,209 @@
+    WD.ZA2   = np.zeros((m,num_prop*(nts-1)*B*n))   
+    WD.XB1   = np.zeros((m,num_prop*(nts-1)*B*n))
+    WD.YB1   = np.zeros((m,num_prop*(nts-1)*B*n))
+    WD.ZB1   = np.zeros((m,num_prop*(nts-1)*B*n))
+    WD.XB2   = np.zeros((m,num_prop*(nts-1)*B*n))
+    WD.YB2   = np.zeros((m,num_prop*(nts-1)*B*n))
+    WD.ZB2   = np.zeros((m,num_prop*(nts-1)*B*n))  
+    
+    
+    VD.Wake = Data()
+    VD.Wake.XA1   = np.zeros((num_prop,(nts-1),B,n))
+    VD.Wake.YA1   = np.zeros((num_prop,(nts-1),B,n))
+    VD.Wake.ZA1   = np.zeros((num_prop,(nts-1),B,n))
+    VD.Wake.XA2   = np.zeros((num_prop,(nts-1),B,n))
+    VD.Wake.YA2   = np.zeros((num_prop,(nts-1),B,n))
+    VD.Wake.ZA2   = np.zeros((num_prop,(nts-1),B,n))   
+    VD.Wake.XB1   = np.zeros((num_prop,(nts-1),B,n))
+    VD.Wake.YB1   = np.zeros((num_prop,(nts-1),B,n))
+    VD.Wake.ZB1   = np.zeros((num_prop,(nts-1),B,n))
+    VD.Wake.XB2   = np.zeros((num_prop,(nts-1),B,n))
+    VD.Wake.YB2   = np.zeros((num_prop,(nts-1),B,n))
+    VD.Wake.ZB2   = np.zeros((num_prop,(nts-1),B,n))    
+     
+    for i in range(num_prop): 
+        Gamma  = np.zeros((m,nts-1,B,n))   
+        num = int(Na/B)  
+        for t_idx in range(nts-1): 
+            for B_idx in range(B): 
+                B_loc = (B_idx*num + t_idx)%Na  
+                Gamma[:,t_idx,B_idx,:] = gamma_new[:,B_loc,:]  
+        
+        #( control point, time step , blade number , location on blade )
+        sx_inf0      = np.multiply(V_prop*np.cos(wake_skew_angle), np.atleast_2d(ts))
+        sx_inf       = np.repeat(np.repeat(sx_inf0[:, :,  np.newaxis], Nr, axis = 2)[:,  : ,np.newaxis,  :], B, axis = 2)  
+                     
+        sy_inf0      = np.multiply(np.atleast_2d(V_inf[:,1]).T,np.atleast_2d(ts)) # = zero since no crosswind
+        sy_inf       = np.repeat(np.repeat(sy_inf0[:, :,  np.newaxis], Nr, axis = 2)[:,  : ,np.newaxis,  :], B, axis = 2)   
+                     
+        sz_inf0      = np.multiply(V_prop*np.sin(wake_skew_angle),np.atleast_2d(ts))
+        sz_inf       = np.repeat(np.repeat(sz_inf0[:, :,  np.newaxis], Nr, axis = 2)[:,  : ,np.newaxis,  :], B, axis = 2)           
+        
+        angle_offset       = np.repeat(np.repeat(np.multiply(omega,np.atleast_2d(ts))[:, :,  np.newaxis],B, axis = 2)[:, :,:, np.newaxis],Nr, axis = 3) 
+        blade_angle_loc    = np.repeat(np.repeat(np.tile(np.atleast_2d(blade_angles),(m,1))[:,  np.newaxis, :],nts, axis = 1) [:, :,:, np.newaxis],Nr, axis = 3) 
+        start_angle_offset = np.repeat(np.repeat(np.atleast_2d(start_angle)[:, :, np.newaxis],B, axis = 2)[:, :,:, np.newaxis],Nr, axis = 3) 
+        
+        # adjust for clockwise/counter clockwise rotation
+        total_angle_offset = angle_offset - start_angle_offset
+        if (prop.rotation != None) and (prop.rotation[i] == 1):        
+            total_angle_offset = -total_angle_offset    
+        
+        azi_y   = np.sin(blade_angle_loc + total_angle_offset) # try positive then negative
```

Comment:
what does this comment mean?

Reply:
Sorry. This was leftover when I was addressing a bug Racheal pointed out

Reply:
Removed